src/server/scripts/EasternKingdoms/ZulGurub/boss_marli.cpp

```diff
@@ -77,142 +81,187 @@ enum Misc
 float const DamageIncrease = 35.0f;
 float const DamageDecrease = 100.f / (1.f + DamageIncrease / 100.f) - 100.f;
 
+float const MaxRangeHatch = 100.f;
+
 class boss_marli : public CreatureScript
 {
     public: boss_marli() : CreatureScript("boss_marli") { }
 
         struct boss_marliAI : public BossAI
         {
-            boss_marliAI(Creature* creature) : BossAI(creature, DATA_MARLI) { }
+            public:
+                boss_marliAI(Creature* creature) : BossAI(creature, DATA_MARLI) { }
 
-            void Reset() override
-            {
-                if (events.IsInPhase(PHASE_THREE))
-                    me->ApplyStatPctModifier(UNIT_MOD_DAMAGE_MAINHAND, TOTAL_PCT, DamageDecrease); // hack
-                _Reset();
-            }
+                void Reset() override
+                {
+                    if (events.IsInPhase(PHASE_THREE))
+                        me->ApplyStatPctModifier(UNIT_MOD_DAMAGE_MAINHAND, TOTAL_PCT, DamageDecrease); // hack
+                    /*std::for_each(_hatchedEggsGuids.begin(), _hatchedEggsGuids.end(), [=](ObjectGuid guid)
+                        {
+                            if (GameObject * egg = ObjectAccessor::GetGameObject(*me, guid))
+                            {
+                                egg->Respawn();
+                            }
+                        });*/
 
-            void JustDied(Unit* /*killer*/) override
-            {
-                _JustDied();
-                Talk(SAY_DEATH);
-            }
+                    summons.DespawnAll();
+                    //_hatchedEggsGuids.clear();
+                    _Reset();
+                }
+                
+                //void SpellHitTarget(GameObject* target, SpellInfo const* /*spellInfo*/) override
+                //{
+                //    if(target->GetEntry() == GOB_SPIDER_EGG)
+                //        _hatchedEggsGuids.insert(target->GetGUID());
```

Comment:
If you intend to stop using part of the code, TC recommends that you remove it instead of commenting it out.

Reply:
prolly PR is draft

Reply:
Would be nice if the Github Pull Request Draft feature had been used to make it clear.

Reply:
How can I use this feature @illfated, kinda new to the github organizational features

Reply:
Sorry for being unclear about "the Github Pull Request Draft feature". That feature is only available when opening a pull request, as a drop-down menu, on the button, where you can choose to publish the PR as a Draft instead of being ready for review (but you can convert the PR to "Ready for review" later).

The main intention here was to point out that it is normally recommended to remove the lines of code instead of commenting it out, at least if there is no important point in keeping the code commented out (if the code needs to be used later).

Reply:
I see, will use it in my future requests, regarding commented code. Problem is that I am unexperienced in scripting yet and how it really should look like is that the boss casts the spell, hatches the egg and after reset event eggs are respawned and all the commented code works on 90% except the respawn part, I tried figuring it out for like 3-4 days debugging but I couldn't so I left it here in case someone knows what might be the reason behind all of this and perhaps point me to right direction, otherwise if maintainers accept this solution as it is I will delete all the unnecessary code before merging :)